service/javadsl/server/src/main/java/com/lightbend/lagom/javadsl/server/ServiceGuiceSupport.java

```diff
@@ -3,23 +3,49 @@
  */
 package com.lightbend.lagom.javadsl.server;
 
-import com.lightbend.lagom.internal.javadsl.*;
+import com.google.inject.Binder;
+import com.lightbend.lagom.internal.javadsl.BinderAccessor;
+import com.lightbend.lagom.internal.javadsl.server.JavadslServicesRouter;
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServices;
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServicesProvider;
 import com.lightbend.lagom.internal.javadsl.server.ServiceInfoProvider;
-import com.lightbend.lagom.internal.javadsl.server.JavadslServicesRouter;
 import com.lightbend.lagom.internal.server.status.MetricsServiceImpl;
-
-import com.lightbend.lagom.javadsl.server.status.MetricsService;
-import com.google.inject.Binder;
 import com.lightbend.lagom.javadsl.api.ServiceInfo;
 import com.lightbend.lagom.javadsl.client.ServiceClientGuiceSupport;
+import com.lightbend.lagom.javadsl.server.status.MetricsService;
+
+import java.util.Arrays;
+import java.util.Optional;
 
 public interface ServiceGuiceSupport extends ServiceClientGuiceSupport {
 
+    /**
+     * @deprecated use {@link ServiceGuiceSupport#bindServices(String, ServiceBinding[])} instead.
+     */
     default void bindServices(ServiceBinding<?>... serviceBindings) {
+        bindServices(Optional.empty(), serviceBindings);
+    }
+
+    /**
+     * binds the provided services and builds the {@link ServiceInfo} for this service.
+     *
+     * @param serviceName
+     * @param serviceBindings
+     */
+    default void bindServices(String serviceName, ServiceBinding<?>... serviceBindings) {
+        bindServices(Optional.of(serviceName), serviceBindings);
+    }
```

Comment:
The service name should come from the "primary" service, I don't think it should be passed explicitly here, that's a duplication.  I think the interface should be:

```java
bindServices(ServiceBinding<?> primaryServiceBinding, ServiceBinding<?> otherServiceBindings)
```

This will also mean that the change is source compatible, so no migration required when upgrading (though not binary compatible, so the other method will still need to be kept).